tests/integration_tests/test_ray.py

```diff
@@ -368,12 +378,16 @@ def test_ray_outputs(dataset_type, ray_cluster_2cpu):
     input_features = [
         binary_feature(),
     ]
+    set_feature_config = set_feature(
+        decoder={"vocab_size": 3},
+        preprocessing={"missing_value_strategy": "fill_with_const", "fill_value": ""},
```

Comment:
We must fill with empty strings for Set output features, otherwise NaNs will be inserted, and the default behavior for NaNs in output features is to `DROP_ROWS`, which leads to poorly handled non-determinism when comparing the metrics between the local and Ray backends.

Reply:
@geoffreyangus Do you want to add that as a comment for this test? Seems useful for anyone who might change this test in the future